buildpacks/_shared/shared/func_yaml.go

```diff
@@ -1,7 +1,7 @@
 // Copyright 2021-2022 VMware, Inc.
 // SPDX-License-Identifier: BSD-2-Clause
 
-package java
+package shared
```

Comment:
If this is not a temporarily generated folder, why are we prefixing it with underscore?

Reply:
Overall, I'm not a fan of the filestructure now being
```
_shared/
python/
java/
```
and think that if a refactor PR was necessary, we should be refactoring into a proper Go Project. Since we broke standards for a per-language folder structure, is building off that the best idea?

Reply:
The underscore was just to differentiate it from the buildpacks, since it's not a buildpack itself. Happy to take suggestions on other names or locations. I considered moving it outside the `buildpacks` dir, but I also wanted it to be clear that it is shared _buildpack_ code, vs general shared code.

Reply:
In Go, that underscore would mean that the folder/file should be ignored. I still personally don't feel like 3 shared functions extracted warrants the generation of an internal dependency, which also feels a bit against Go, but alas I'll let others comment for approving.

Reply:
Gotcha. Didn't realize that about the underscore. Have an alternative suggestion?

For the shared functions, I would focus on the usage of the individual functions rather than the total number of them. There may be just a handful of small shared functions, but for each one we are using them for N buildpacks.

Reply:
[This article](https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common) summarizes kinda what I'm getting at. Golang's [direct blog](https://go.dev/blog/package-names?utm_source=dlvr.it&utm_medium=twitter) also says to directly avoid this. We basically have a `utils` or `helpers` folder now sitting, so we're either breaking repetition or we're breaking structure.

Definitely let's remove the `_` underscore prefix. If I _had_ to name it, I would follow their guidance and name the package relevant to what it does: it handles `func.yaml`.

Reply:
Ah, I see what you're getting at. I'm with you on package names for sure and will address this.

That said, the package name is a separate concern from the folder (and/or module) name that houses these shared packages, so what do you think of this?

```
buildpacks/
    common/
        go.mod  # module name "kn-fn/buildpacks", per later thread in this PR
        func/  # package to house FuncYaml
        command/  # package to house `command.Runner` interface and default implementation
        ...  # other packages as necessary over time
    python/
    java/    
```

cc: @andrew-su @joeeltgroth

Reply:
I think either `common` or `shared` is fine (without the `_`). The readme in this folder could be used to hold the context on which the folders are. Would you be okay with pulling this into its own PR? I see your comment further down about DI. Could this be something that is added to the shared folder as a followup refactor?

Reply:
Sounds good. I'll spend some time to simplify this PR and will spin out further stories/PRs for the other work